exp/zapslog/slog.go

```diff
@@ -25,10 +25,46 @@ import (
 	"runtime"
 
 	"go.uber.org/zap"
+	"go.uber.org/zap/internal/pool"
 	"go.uber.org/zap/zapcore"
 	"golang.org/x/exp/slog"
 )
 
+type fieldSlice []zapcore.Field
+
+func (s *fieldSlice) Append(f zapcore.Field) {
+	*s = append(*s, f)
+}
+
+func (s *fieldSlice) AddTo(enc zapcore.ObjectEncoder) {
+	for i := 0; i < s.Len(); i++ {
+		(*s)[i].AddTo(enc)
+	}
```

Comment:
Any reason not to use `range` here?
```suggestion
	for _, f := range (*s) {
		f.AddTo(enc)
	}
```